cli/kaos_cli/commands/build.py

```diff
@@ -28,94 +42,201 @@
-                click.style(env, bold=True, fg='blue')))
+                click.style(env_state.cloud, bold=True, fg='red'),
+                click.style(env_state.env, bold=True, fg='red'))),
+            sys.exit(1)
+
+    except Exception as e:
+        handle_specific_exception(e)
+        handle_exception(e)
+
+
+# build list
+# =============
+@build.command(name='list', short_help='List currently deployed backend builds')
+@pass_obj(BackendFacade)
+def list_all(backend: BackendFacade):
+    """
+    List all deployed kaos backend infrastructures.
+    """
+    try:
+        available_contexts = backend.list()
+
+        if available_contexts:
+            backend.cache(available_contexts)
+            table = render_table(available_contexts, include_ind=False)
+            click.echo(table)
         else:
-            click.echo("{} - Successfully built {} environment".format(
-                click.style("Info", bold=True, fg='green'),
-                click.style('kaos', bold=True)))
+            click.echo("{} - No active builds found. Please run {} to deploy an environment".format(
+                click.style("Warning", bold=True, fg='yellow'),
+                click.style('kaos build deploy', bold=True, fg='green')))
+
+    except Exception as e:
+        handle_specific_exception(e)
+        handle_exception(e)
+
+
+# build set
+# =============
+@build.command(name='set', short_help='Set active build context')
+@click.option('-c', '--context', type=str, help='set to a specific deployed context', cls=NotRequiredIf,
+              not_required_if='ind')
+@click.option('-i', '--ind', type=int, help='set to a specific deployed index', cls=NotRequiredIf,
+              not_required_if='context')
+@pass_obj(BackendFacade)
+def set_active_context(backend: BackendFacade, context: Optional[str] = None, ind: Optional[int] = None):
+    """
+    Set current model environment workspace.
+    """
+    try:
+        # ensure arguments are correctly defined
+        validate_inputs([context, ind], ['context', 'ind'])
+        # selection by index
+        if context:
+            is_context_set = backend.set_context_by_context(context)
+            if not is_context_set:
+                click.echo('Context {} invalid. It is not one of the existing deployments in {} '
+                           .format(click.style(context, bold=True, fg='red'), click.style("kaos", bold=True)))
+                sys.exit(1)
+            click.echo("{} - Successfully set to context - {}".
+                       format(click.style("Info", bold=True, fg='green'), click.style(context, bold=True, fg='green')))
+        if ind or ind == 0:
+            is_context_set, context = backend.set_context_by_index(ind)
+            if not is_context_set:
+                click.echo('Index {} invalid. It is not one of the existing deployments in {} '
+                           .format(click.style(context, bold=True, fg='red'), click.style("kaos", bold=True)))
```

Comment:
This line is causing the `set -i <INVALID INDEX>` bug to happen, `context` variable is None and you need to display the `ind` variable instead:
```
.format(click.style(str(ind), bold=True, fg='red'), click.style("kaos", bold=True)))
```